### PR TITLE
Added support to send logs from preview-environment to GCP

### PR DESCRIPTION
**Pull request by wulfthimm**
## Description

This adds `fluent-bit` to preview-environment VMs sending logs to GCP.

## Related Issue(s)

Fixes https://github.com/gitpod-io/ops/issues/640

## How to test

- checkout a new branch
- push a commit creating a new VM
- check GCP logging by searching for `jsonPayload.kubernetes.host="<NAMESPACE>"`

The NAMESPACE is the normalized branch name like we already do it for domains. In this case it is "wth-prev-log".

## Release Notes

```release-note
None
```

### CODE REVIEW

**Comment by roboquat** [comment]
@wulfthimm: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jankoehnlein** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jankoehnlein` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: .werft/build.ts
Comment (ArthurSens) [inline .werft/build.ts:357]:
```suggestion
        VM.installFluentBit({namespace: 'default', slice: vmSlices.EXTERNAL_LOGGING});
        werft.done(vmSlices.EXTERNAL_LOGGING)
```

It's important to close slices so we get nice traces in honeycomb :) 
See also: https://github.com/gitpod-io/gitpod/pull/7984

File: .werft/values.fluent-bit.yaml
Comment (ArthurSens) [inline .werft/values.fluent-bit.yaml:20]: Since we have some manifests in typescript for harvester stuff, maybe this manifest could also live in `.werft/vm/manifests.ts`?

I believe it'd be nice if we agree on one single way to handle manifests, but no strong opinions!

@mads-hartmann what do you think?

**Review by ArthurSens — COMMENTED** [review]
Haven't tested yet, just skimming through the changes :)

File: .werft/values.fluent-bit.yaml
Comment (wulfthimm) [reply .werft/values.fluent-bit.yaml:20]: I thought `manifests.ts` is for actual resources. This is only a values file like `values.tracing.yaml` which is also in `.werft`.

Comment (mads-hartmann) [reply .werft/values.fluent-bit.yaml:20]: I think this might have to be a `yaml` file as it is passed to helm using `-f` but I'd very much like for it not to be directly in the `.werft` folder ([see RFC](https://www.notion.so/gitpod/Decomposing-the-build-job-2327d3cd8dfa4520b0ad994ab1b49dfd)).

@wulfthimm How about moving this to somewhere else? Maybe dev/charts/fluentbit or .werft/vm/charts/fluentbit.

Alternatively (and maybe this was what you are suggestion @ArthurSens) we could have it be part of the user data and install it during boot that way (in `.werft/vm/manifests.ts`)

File: .werft/build.ts
Comment (ArthurSens) [inline .werft/build.ts:358]:
```suggestion
        exec(`kubectl apply -f clouddns-dns01-solver-svc-acct.yaml -f letsencrypt-issuer.yaml`, { slice: vmSlices.INSTALL_LETS_ENCRYPT_ISSUER, dontCheckRc: true })
        werft.done(vmSlices.INSTALL_LETS_ENCRYPT_ISSUER)

        VM.installFluentBit({namespace: 'default', slice: vmSlices.EXTERNAL_LOGGING});
        werft.done(vmSlices.EXTERNAL_LOGGING)
        
```

Sorry, the first suggestion messed up the order a little bit 😅

File: .werft/values.fluent-bit.yaml
Comment (ArthurSens) [reply .werft/values.fluent-bit.yaml:20]: I wasn't thinking about boot, but I like that idea even better

Comment (wulfthimm) [reply .werft/values.fluent-bit.yaml:20]: I do not like the idea to add it to the boot process as the serviceAccount secret should be installed before the installation of `fluent-bit`.

**Comment by ArthurSens** [comment]
I just created a new branch called 'as/test-vm-log' and I'm trying to search logs from GCP with the query `jsonPayload.kubernetes.host=as/test-vm-log`. Nothing is being returned 🤔 


Do I need to be in a specific gcp project? Is the query I'm using correct?

**Comment by wulfthimm** [comment]
> I just created a new branch called 'as/test-vm-log' and I'm trying to search logs from GCP with the query `jsonPayload.kubernetes.host=as/test-vm-log`. Nothing is being returned thinking
> 
> Do I need to be in a specific gcp project? Is the query I'm using correct?

The NAMESPACE is the normalized branch name like we already do it for domains. In this case it is "wth-prev-log".

**Review by ArthurSens — APPROVED** [review]
Noice 👍, it works when I follow your suggestion!

I believe we should have some follow-up tasks regarding logging in previews, but let's get this merged!

Maybe worth to document future tasks:
* Revisit and investigate fluent-bit configuration, we may want to add/exclude attributes so devX can be improved
* We need to document this somewhere so the engineering teams also have the necessary knowledge to search for their preview's logs

cc @mads-hartmann

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 36b0349fba136bd1f431c1a76a40ac38ea828c0a</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/8011#pullrequestreview-872026831" title="Approved">ArthurSens</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jankoehnlein** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jankoehnlein` in a comment when ready.

Associated issue: *#640*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/8011#pullrequestreview-872026831" title="Approved">ArthurSens</a>*

Associated issue: *#640*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
